### PR TITLE
[4.0] Namespace com_contenthistory

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This simple PR converts com_contenthistory component to namespace MVC for Joomla 4.

### Testing Instructions
1. Install latest Joomla 4.0-dev with sample data, then apply patch (or better install my branch https://github.com/joomdonation/joomla-cms/archive/namespace_com_contenthistory.zip )

2. Login to administrator  Content -> Article, try to edit an article, save it (so that version of that version history of that article saved)

3. Try to edit that article again, click on Versions button, there will be a popup displays version history of that article. Check the Restore, Compare, Preview... , make sure it is still working as before

### Additional comment
When all components namespaced , we need to review and update data for #__content_types table, table field, data for prefix setting (currently, it is JTable, need to correct it to correct table prefix)

### CODE REVIEW

**Comment by joomdonation** [comment]
Drone error causes by something else and not related to this PR

File: administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+					$object->$name = $subObject;
+				}
+			}
+		}
+
+		return $object;
+	}
+
+	/**
+	 * Method to get field labels for the fields in the \JSON-encoded object.
```
Comment (brianteeman) [inline administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php]: it looks wrong to have \JSON in the comment is that a search replace error adding the \

File: administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+					$result[$subName] = $subValue;
+				}
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Method to decode \JSON-encoded fields in a standard object. Used to unpack \JSON strings in the content history data column.
```
Comment (brianteeman) [inline administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php]: it looks wrong to have \JSON in the comment is that a search replace error adding the \

File: administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+	 *
+	 * @param   \stdClass  $jsonString  Standard class object that may contain one or more \JSON-encoded fields.
+	 *
+	 * @return  \stdClass  Object with any \JSON-encoded fields unpacked.
```
Comment (brianteeman) [inline administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php]: it looks wrong to have \JSON in the comment is that a search replace error adding the \

**Review by brianteeman — COMMENTED** [review]
it looks wrong to have \JSON in the comment is that a search replace error adding the \

**Comment by wilsonge** [comment]
Thanks guys :)

**Pull request merged**